crossplane/objects.py

```diff
@@ -0,0 +1,202 @@
+# -*- coding: utf-8 -*-
+from .parser import parse
+
+
+#
+# Functions
+#
+
+def _init_directive(directive_json):
+    return NGXBlockDirective(**directive_json) if 'block' in directive_json \
+        else NGXDirective(**directive_json)
+
+
+#
+# Classes
+#
+
+class NGXDirective(object):
+    __slots__ = ('directive', 'line', 'args', 'includes')
+
+    def __init__(self, directive='', line=0, args=[], includes=[], **kwargs):
+        self.directive = directive
+        self.line = line
+        self.args = args
+        self.includes = []
+
+    def dict(self):
+        return {
+            slot: getattr(self, slot)
+            for slot in self.__slots__
+        }
+
+
+class NGXBlockDirective(NGXDirective):
+    __slots__ = ('index', 'directive', 'line', 'args', 'block')
+
+    def __init__(self, block=[], **kwargs):
+        super(NGXBlockDirective, self).__init__(**kwargs)
+        self.index = {}
+        self.block = block
```

Comment:
`self.block` shouldn't be set to `block` (the list of dict) if it's going to be reset to a dict of Python objects after `_setup_block`.

Reply:
It gets stored as an attribute as passed and then converts each element into Python objects after.  You could do it all in the same step, but I was thinking that it would be easier to bubble more descriptive errors in the future.  E.G. was the element missing from the payload vs did something go wrong in conversion.